arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -1009,111 +1014,150 @@ Result DatabaseInitialSyncer::fetchCollectionSyncByKeys(arangodb::LogicalCollect
-      stats.waitedForInitial += waitTime;
-      return Result(TRI_ERROR_REPLICATION_NO_RESPONSE,
-                    std::string(
-                        "timed out waiting for response from leader at ") +
-                        _config.leader.endpoint);
+      return replutils::buildHttpError(response.get(), url, _config.connection);
     }
 
-    if (isAborted()) {
-      stats.waitedForInitial += waitTime;
-      return Result(TRI_ERROR_REPLICATION_APPLIER_STOPPED);
+    VPackBuilder builder;
+    Result r = replutils::parseResponse(builder, response.get());
+
+    if (r.fail()) {
+      ++stats.numFailedConnects;
+      return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
+                    concatT("got invalid response from leader at ",
+                            _config.leader.endpoint, url, ": ", r.errorMessage()));
     }
 
-    std::chrono::milliseconds sleepTime = ::sleepTimeFromWaitTime(waitTime);
-    std::this_thread::sleep_for(sleepTime);
-  }
-  
-  stats.waitedForInitial += TRI_microtime() - startTime;
+    VPackSlice const slice = builder.slice();
+    if (!slice.isObject()) {
+      ++stats.numFailedConnects;
+      return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
+                    std::string("got invalid response from leader at ") +
+                    _config.leader.endpoint + url + ": response is no object");
+    }
 
-  if (replutils::hasFailed(response.get())) {
-    ++stats.numFailedConnects;
-    return replutils::buildHttpError(response.get(), url, _config.connection);
-  }
+    return Result();
 
-  VPackBuilder builder;
-  Result r = replutils::parseResponse(builder, response.get());
+  };
 
-  if (r.fail()) {
-    ++stats.numFailedConnects;
+  auto ck = keysCall(true);
+  if (!ck.ok()) {
+    return ck;
+  }
+  VPackSlice const c = slice.get("count");
+  uint64_t ndocs = 0;
+  if (c.isNumber()) {
+    ndocs = c.getNumber<uint64_t>();
+  } else {
     return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
-                  concatT("got invalid response from leader at ",
-                          _config.leader.endpoint, url, ": ", r.errorMessage()));
+                  std::string("got invalid response from master at ") +
+                  _config.leader.endpoint + url + ": response count not a number");
   }
 
-  VPackSlice const slice = builder.slice();
-  if (!slice.isObject()) {
-    ++stats.numFailedConnects;
-    return Result(TRI_ERROR_REPLICATION_INVALID_RESPONSE,
-                  std::string("got invalid response from leader at ") +
-                      _config.leader.endpoint + url + ": response is no object");
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+  if (ndocs > _quickKeysNumDocsLimit && slice.hasKey("id")) {
+    LOG_TOPIC("6e1b3", ERR, Logger::REPLICATION)
+        << "client: DatabaseInitialSyncer::run - expected ony document count for quick call";
+    TRI_ASSERT(false);
+  }
+#endif
+
+  if (!slice.hasKey("id")) { // we only have count
+    maxWaitTime = ndocs * 8 / 100000;
```

Comment:
Isn't this doing integer division, so for any number of documents < 12500, the maxWaitTime is 0? I don't think this is desired, and I suggested some fix before to not make the calculated wait time go below what was configured in `_initialSyncMaxWaitTime`.
I also don't understand why we multiply with 8 first. What is the reasoning here? A code comment would help and clarify it.

Reply:
absolutely. fixed.